src/cli.rs

```diff
@@ -354,10 +373,23 @@ pub enum LogFormat {
     Json,
 }
 
+#[derive(clap::ValueEnum, Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ConfigWatcher {
+    Inotify,
+    Poll,
+}
+
 pub fn handle_config_errors(errors: Vec<String>) -> exitcode::ExitCode {
     for error in errors {
         error!(message = "Configuration error.", %error);
     }
 
     exitcode::CONFIG
 }
+
+pub fn watcher_config(method: ConfigWatcher, interval: NonZeroU64) -> watcher::ConfigWatcher {
+    match method {
+        ConfigWatcher::Inotify => watcher::ConfigWatcher::RecommendedWatcher,
```

Comment:
the `ConfigWatcher::Inotify` variant does not always set the `inotify` watcher, it sets the `RecommendedWatcher`, which would be `inotify` on supported platforms and polling in others.

I think we should rename those variants (or at least, the `Inotify` one), as it is a bit misleading for the user (also, the default value in the args should be change accordingly)